src/workerd/api/memory-cache.c++

```diff
@@ -449,13 +449,15 @@ jsg::Promise<jsg::JsRef<jsg::JsValue>> MemoryCache::read(jsg::Lock& js,
   }
 }
 
-void MemoryCache::delete_(jsg::Lock& js, jsg::NonCoercible<kj::String> key) {
-  // Ignore operations on keys exceeding key max size.
+jsg::Promise<void> MemoryCache::delete_(jsg::Lock& js, jsg::NonCoercible<kj::String> key) {
   if (key.value.size() > MAX_KEY_SIZE) {
-    return;
+    return js.rejectedPromise<void>(js.rangeError("Key too large."_kj));
   }
 
+  auto deleteSpan = IoContext::current().makeTraceSpan("memory_cache_delete"_kjc);
```

Comment:
I'm not very familiar with volatile cache/memory cache, but except for acquiring the lock in `SharedMemoryCache::Use::delete_()` this looks like a mostly synchronous operation here right? Not sure if we should add a span here since we'd expect the operation to complete very quickly; we'd also need to add some internal plumbing to support/document the span.